src/main/java/org/folio/sample/controller/BookController.java

```diff
@@ -68,12 +67,35 @@ public ResponseEntity<BookDTO> getBook(UUID bookId) {
     log.info("Called GET /books/{}", bookId);
 
     return new ResponseEntity<>(
-      bookMapper.toDto(
+        bookMapper.toDto(
+            bookService
+                .getBookById(bookId)
+                .orElseThrow(() -> new NotFoundException("Book not found"))),
+        HttpStatus.OK);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public ResponseEntity<BookDTO> updateBook(UUID bookId, BookForCreationDTO book) {
+    log.info("Called PUT /books/{}", bookId);
+
+    // Get params for new book
+    String newBookName = book.getName();
+    LocalDate newBookPublishedDate = book.getPublishedDate();
+
+    // Get old book, throw exception if not found and respond 404
+    BookDTO oldBook = bookMapper.toDto(
```

Comment:
Any particular reason you're converting the database entity to the API model, just to convert it back on save? Seems like it might make more sense to go the other way around.  There's a fromDto method on the mapper that might be useful for this (see the createBook implementation); it converts BookDTO to a Book (and just leaves the ID null)